ODIN_II/parse_odin_stats.py

```diff
@@ -0,0 +1,948 @@
+            ratioLine += "INF" if(percentChange=="INF") else "0" if(diff==0) else "-100.0" if float(symbolA) and not float(symbolB) else str(round((diff / float(symbolA)),4)*100)
+            ratioLine += delim
+            if(thresholds and thresholds.get(headerVal) != None):
+                if(float(symbolB) == 0 or float(symbolA) == 0):
+                    withinLimit = "N/A"
+                else:
+                    withinLimit = within_threshold(thresholds, headerVal, diff, float(percentChange))
+            else:
+                withinLimit = "N/A"
+            isFlagged = is_flagged(withinLimit, float(symbolB), float(symbolA))
+            flagged = "No"
+            if(isFlagged):
+                flagged = "Yes"
+                flags += 1
+                thresholdCoefficient = 0.0
+                if(thresholds and thresholds.get(headerVal) != None):
+                    thresholdCoefficient = float(thresholds.get(headerVal)) / 100
+                flagLine += flag_line(headerVal, withinLimit, flags, diff, thresholdCoefficient, float(symbolA), percentChange)
+            percentChange +="%"+" "*(10-len(percentChange))
+            if(not conciseOutput):
+                print("\t"+headerVal+":\t\t\t"+spacing+sign+str(diff)+"\t\t\t"+str(ratio)+"\t\t"+percentChange+biggerSmaller+" \t"+withinLimit+"\t\t\t"+flagged)
+        else:
+            ret_str += "NaN"+delim
+            ratioLine += "NaN"+delim if(printRatios) else ""
+        index += 1
+        headerVal = str(header_in.split(delim)[index])
+    if(flags >= 1):
+        output_flagged(flagLine, flags, input1_name, generate_flag_files)
+    ret_str += ("\n"+ratioLine) if(printRatios) else ""
+    return ret_str
+    
+
+def insert_string(value_map, key, input_str):
+    value_map[key] = format_token(input_str)
+
+def insert_decimal(value_map, key, input_str):
+    value_map[key] = str(assert_float(format_token(input_str)))
+
+# quickly generates thresholds for core stats, same value for each stat
+def generate_thresholds(percentage, filename):
+    ft = open(filename,"w+")
+    wrt = ""
+    coreStats = ["num_nodes"
+        , "num_names"
+        , "num_latches"
+        , "num_edges_fo"
+        , "num_input_nodes"
+        , "num_top_output_nodes"
+        , "max_dist_out"
+        , "min_dist_out"
+        , "num_already_visited"
+        , "top_input_avg_dist_min"
+        , "top_input_avg_dist_max"
+        , "longest_minpath"
+        , "shortest_maxpath"
+        , "vcc_fanout_live"
+        , "gnd_fanout_live"]
+        #, "nodeOutputCount"]
+    for header in coreStats:
+        wrt += header + ": "+str(percentage)+"\n"
+    ft.write(wrt)
+
+# TODO the parser really doesn't like newline after word w/out space, which could cause problems for people creating their own threshold files
+def parse_line(benchmarks, line):
+
+    line.strip()
+    line.strip("\n")
+    line = " ".join(line.split())
```

Comment:
I think that simply doing ```line = " ".join(line.split()``` should achieve the same thing and fix your TODO above